src/corelibs/U2Formats/src/PDBFormat.cpp

```diff
@@ -426,7 +426,11 @@ void PDBFormat::PDBParser::parseAtom(BioStruct3D &biostruct, U2OpStatus &) {
     biostruct.modelMap[currentModelIndex + 1].insert(id, a);
 
     if (atomIsInChain) {
+        SAFE_POINT_EXT(0 <= chainIndex && chainIndex < biostruct.moleculeMap.size(), os.setError("Unexpected chainIndex"), );
+
         SharedMolecule &mol = biostruct.moleculeMap[chainIndex];
+        SAFE_POINT_EXT(0 <= currentModelIndex && currentModelIndex < mol->models.size(), os.setError("Unexpected currentModelIndex"), );
```

Comment:
Using Yoda-style comparisons makes code unreadable.

Our build has a special compiler flag (see ugene-globals.pri) that will make assignment in expressions a compile-time error.

Reply:
As far as I'm concerned, this is the most convenient way for the double comparisons visualizing. Let's remember, how it was in school:
a <= X < b
From the left to the right, the left number is the smallest one, the right number is the largest one, the variable in the middle. Unfortunately, C++ doesn't support double comparisons as one logical action, so the following code is invalid:
`0 <= currentModelIndex < mol->models.size()`
But, we can use LOGICAL AND and transfer the above-mentioned code to the most similar one:
`0 <= currentModelIndex && currentModelIndex < mol->models.size()`

The second part isn't clear for me.